packages/core/test/stateIn.test.ts

```diff
@@ -424,4 +424,44 @@ describe('transition "in" check', () => {
     actorRef.send({ type: 'TIMER' });
     expect(actorRef.getSnapshot().value).toEqual('green');
   });
+
+  it('should allow a guard object to reference another guard object', () => {
```

Comment:
I'd move this test to a new test dedicated to `stateIn`, something like:
```suggestion
  it('should be possible to use a referenced `stateIn` guard', () => {
```

It's an important, basic~, case that needs to work and I think it makes sense to have a specific test for it.

The current test title is good as well, I'd just move it to a different test suite (since this one is in `stateIn.tes.ts`) and use some random~ custom guards in it.